pkg/statsd/aggregator.go

```diff
@@ -284,20 +287,23 @@ func (a *MetricAggregator) receiveGauge(m *gostatsd.Metric, tagsKey string, now
 
 func (a *MetricAggregator) receiveTimer(m *gostatsd.Metric, tagsKey string, now gostatsd.Nanotime) {
 	v, ok := a.Timers[m.Name]
-	if ok {
-		t, ok := v[tagsKey]
-		if ok {
-			t.Values = append(t.Values, m.Value)
-			t.Timestamp = now
-		} else {
-			t = gostatsd.NewTimer(now, []float64{m.Value}, m.Hostname, m.Tags)
+	if !ok {
```

Comment:
This restructuring changes it away from the other 3 ```receive*``` calls.  I generally agree that it's a cleaner pattern, however historically whenever I've done this, there's a performance impact (not a lot, but present) that I can't justify just for the style change.

Reply:
Fixed, reverted `receiveTimer` to original structure